src/item.cpp

```diff
@@ -2142,6 +2148,15 @@ void item::on_pickup( Character &p )
     }
 }
 
+void item::on_contents_changed()
+{
+    if( !is_non_resealable_container() ) {
+        return;
+    }
+
+    convert( type->container->unseals_into );
```

Comment:
Would invert the conditional and place this within. Later you might want to perform more tasks in this callback and that would cut down the future delta.